### DISCUSSION TITLE
`invalidateQueries` behaviour

### DISCUSSION THREAD

**Discussion by Slmii** [TanStack/query#7327] [Q&A] [answered]
So a question about `invalidateQueries`, cause I am not sure of thats the way it works or not.

Lets say I have a `query` in `Component A`, and I navigate to `Page B`, where `Component A` is not visible/mounted.
I run `invalidateQueries` on the query from `Component A` in `Component B` and navigate back to the page that mounts `Component A`.

What I thought would happen is that when running `invalidateQueries` anywhere the query would be invalidated/refetched  in the background the moment it runs, but what happens instead is, the query invalidates/refetches the moment when `Component A` mounts.

Is this expected behaviour?

**Comment by TkDodo** [comment, marked as answer]
yes, because invalidation will:
- refetch all active / currently used queries
- will mark all other (inactive) queries as stale so that they can be refetched the next type they are used.

if you want to always refetch, use `refetchQueries` instead
